app/src/main/java/com/android/sample/model/workout/WorkoutRepositoryFirestore.kt

```diff
@@ -226,7 +251,30 @@ open class WorkoutRepositoryFirestore<T : Workout>(
         .document(id)
         .delete()
         .addOnFailureListener { e -> onFailure(e) }
-        .addOnSuccessListener { onSuccess() }
+        .addOnSuccessListener {
+          onSuccess()
+          removeWorkoutFromCache(id) // Remove from cache
+        }
+  }
+
+  private fun saveWorkoutsToCache(workouts: List<T>) {
+    CoroutineScope(Dispatchers.IO).launch { localCache.saveWorkouts(workouts) }
+  }
+
+  private fun updateWorkoutInCache(workout: T) {
+    CoroutineScope(Dispatchers.IO).launch {
+      val currentCache = localCache.getWorkouts().firstOrNull() ?: emptyList()
+      val updatedCache = currentCache.map { if (it.workoutId == workout.workoutId) workout else it }
```

Comment:
It comes back to what was discussed with the coaches but is it too complicated to retrieve and modify only the concerned workout.

Reply:
I tried doing it but the thing is from what I understood, the coaches said it would be better to implement that in the user account data where the number of keys is already determined by just being the different fields about the user however here the number of workouts keeps increasing indefinitely and so i thought maybe it would get harder to then fetch all workouts by going through all keys one by one and would be more efficient to save them all in a single place